app/code/Magento/Review/Observer/PredispatchReviewObserver.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Review\Observer;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\UrlInterface;
+
+/**
+ * Class PredispatchReviewObserver
+ *
+ * @package Magento\Review\Observer
+ */
+class PredispatchReviewObserver implements ObserverInterface
```

Comment:
Is better to use a Plugin for that.

https://devdocs.magento.com/guides/v2.2/ext-best-practices/extension-coding/observers-bp.html

Reply:
@larsroettig I think in this case an Observer is the better choice, because it listens to every "review" frontend route.

Reply:
I think observer is more performant solution in this case, the implementation is similar to approved https://github.com/magento/magento2/pull/18407

Reply:
Okay good to know for me the observer pattern is deprecated and should only use to get informed if some event happens but not to change something in the Request Infomation because to find them can be not so easy.

Reply:
So I leave it as it is.

Reply:
@larsroettig you are also right, it is definitely not the best practice to use observers for changing the application flow, because it makes it very hard to debug. However, considering the things described in my previous comment, it's hard for me to decide on the best approach in this case. Also, I think that this observer is introducing so radical/explicit changes to the behaviour (redirect based on "Enabled" config setting) that it might not be too confusing for developers who will be debugging the behaviour..

Reply:
@sivaschenko okay than observer also okay for me mostly I try to apply the Magento 2 best praxis to the Pull Request. Sometimes I know Observer is a better way to do things fast 👍